src/encoder.rs

```diff
@@ -452,10 +452,16 @@ impl<'a, W: Write> Encoder<'a, W> {
 /// to guarantee the next image to be prefaced with a fcTL-chunk, and all other chunks would be
 /// guaranteed to be `IDAT`/not affected by APNG's frame control.
 pub struct Writer<W: Write> {
+    /// The underlying writer.
     w: W,
+    /// The local version of the `Info` struct.
     info: PartialInfo,
+    /// Global encoding options.
     options: Options,
-    written: u64,
+    /// The total number of image frames, counting all consecutive IDAT and fdAT chunks.
+    /// Note: we use `u64::MAX` as a flag to note when the IEND chunk was also added.
```

Comment:
Could we just add an extra bool field instead of using u64::MAX as a sentinel value?